.github/workflows/go.yaml

```diff
@@ -0,0 +1,31 @@
+name: Golang SDK
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  build:
+    env:
+      TEST_ARGS: -v
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.22
+
+      - name: Build and Test
+        run: cd zkp/golang && make
+
+      - uses: codecov/codecov-action@v4
+        with:
+          codecov_yml_path: ./codecov.yml
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
awesome, I have touched based in the pass with the Hyperledger team to setup codecov! Can reach out to set it up for this repo